presto-main/src/test/java/io/prestosql/type/TestArrayOperators.java

```diff
@@ -749,6 +750,23 @@ public void testSubscript()
         assertDecimalFunction("ARRAY [2.22222222222222222, 2.3][1]", decimal("2.22222222222222222"));
     }
 
+    @Test
+    public void testSubscriptReturnType()
+    {
+        // Test return type of array subscript by passing it to another operation
+        // One test for each specialization of the operator
+        assertSubscriptCast("true", "true"); // boolean
+        assertSubscriptCast("1234", "1234"); // long
+        assertSubscriptCast("1.23", "1.23"); // double
+        assertSubscriptCast("IPADDRESS '127.0.0.1'", "127.0.0.1"); // slice
+        assertSubscriptCast("TIMESTAMP '1970-01-01 00:00:00.000000001'", "1970-01-01 00:00:00.000000001"); // object (LongTimestamp)
```

Comment:
a `Block` is a special case.

it works today, dunno why
```
assertFunction("CAST((ARRAY[ARRAY['xxx']])[1] AS json)", JSON, "[\"xxx\"]"); // block
```

but still worth having a test

note: cast here is to JSON, because of https://github.com/trinodb/trino/issues/6347
this anomaly in test cases leads to a question whether `assertSubscriptCast` is worth having separate or inline back. I'd inline.

Reply:
Or instead, I could use JSON for all of them. :eyes: